### ISSUE TITLE
Subscribe to Views / Multiple Rows / Multiple Columns

### ISSUE THREAD

**Issue by jdgamble555** [supabase/realtime-js#185]

**Comment by w3b6x9** [comment]
@jdgamble555 

> Channel names could be automatic (perhaps with UUID in the background) if we don't want to select a channel.

Yes, we can generate a unique string on the client's behalf if one is not provided.

> These classes and methods are MORE confusing, and don't help basic use cases.

Would love to hear more about this one. What do you mean by these classes and methods? How does it not support basic use cases?

We have discussed queries/views/graphql subscriptions internally and we'll explore it further before we prioritize it. I'll post updates here as we continue to discuss internally.

**Comment by jdgamble555** [comment]
> Would love to hear more about this one. What do you mean by these classes and methods? How does it not support basic use cases?

The `.on` and `.channel` with the complex parameters where there is no documentation about options.
___
What seems easier to understand:

**1**
```typescript
supabaseClient
  .channel('any_string_you_want')
  .on(
    'postgres_changes',
    {
      event: 'INSERT',
      schema: 'public',
      table: 'movies',
    },
    (payload) => {
      console.log(payload)
    }
  )
  .subscribe()
```

**2**
```typescript
const mySubscription = supabase
  .from('*')
  .on('*', payload => {
    console.log('Change received!', payload)
  })
  .subscribe()
```

I understand the broadcasting feature seems pretty cool to have.  However, you have over complicated the subscription process for people who just want to subscribe to the database.  I don't know that the broadcasting feature is really important or useful to most people when you could just store items in the database directly and subscribe to that.  Perhaps a good compromise would be to add another method that emulates the promise method for postgres.  This perhaps would call the unintuitive method #1 that you have created for supabase-js-v2.

In most cases, I don't need to think about a public schema, I don't need to think about a channel name, and I don't need to think about an event.  I want to subscribe to:

1. All changes
2. The public schema
3. In postgres
4. And I don't want to name a channel

The broadcasting feature is pretty cool, but we still don't have the NEEDED features with postgres subscriptions.

1. There is a [delay now in subscriptions](https://github.com/supabase/realtime/issues/282) !!! This should be fixed ASAP!
2. We cannot filter any of our subscriptions except for a simple `eq` filter? Firebase straight up can do more than this. A lot of your clients are moving from Firebase for the relational data support, and you can't do relational data with real time subscriptions?
    a. Who decided to add the `eq` clause as a random string in a filter `filter: 'body=eq.hey'`? Why? Why would you do that, especially when you just wrote a new sdk version...
    b. Subscriptions should not work differently than postgres promises.  In fact, I suggest you separate the broadcasting from the postgres subscriptions all together, as they are two different things. This would also solve #1. But more importantly, let's add all the filters that promises have shall we!
3. Besides a real hack of using RLS policies in certain circumstances, there is no real way to work around not being able to subscribe to your actual query.  I don't want to subscribe to all my data.  I only want to subscribe to my query.  I don't need all inserts where x = y.  I need a subset of those inserts, or any changes for that matter.  There is no work around for that, and you guys are still `talking about` why subscribing to `view` changes are appropriate?  This is HUGE! 

Yes, web sockets in Supabase for the moment are very very limiting, and you guys don't seem to have the fixes on your roadmap just yet.

Let's change that please.

J

**Comment by jfreyberg** [comment]
Are there any updates regarding these issues?
@jdgamble555 made some _very_ valid points. The realtime feature needs filtering in order to make sense for most real-world use cases. I find it confusing and somewhat concerning that this is not a top priority, as it clearly should be, while other, much less relevant features are pushed forward.

**Comment by jdgamble555** [comment]
It looks like they [may have added](https://twitter.com/dshukertjr/status/1593094356969590786) the main six: `eq`, `neq`, `gt`, `gte`, `lt`, `lte`.  Unfortunately there is no `or` filter yet, and you are limited to only one filter.

I have added it to my [j-supabase](https://code.build/p/GZ6ioN6YzcpDwNwGNnDpEn/supabase-subscriptions-just-got-easier) package, although I admit, I have not tested this yet.

But again, we need ALL filters that are available on JS sdk, and the ability to use more than one.  I am taking this as HOPE that this is becoming a priority, as it should.

J

**Comment by jfreyberg** [comment]
That surely is a step in the right direction. However without `and` and `or` functionality filtering is not viable for most use cases.

@w3b6x9 (or any other maintainer) could you please give an update on the status of this feature?

**Comment by w3b6x9** [comment]
@jdgamble555 this is good feedback here: https://github.com/supabase/realtime-js/issues/185#issuecomment-1258851450. Will look it over and pass it on to the rest of the team. Also, what do you mean by `postgres promises`? Like the ability to perform db operations via PostgREST and get back a response on promise resolve in JS?

> It looks like they [may have added](https://twitter.com/dshukertjr/status/1593094356969590786) the main six: eq, neq, gt, gte, lt, lte

Yes, these filters are available.

We plan to integrate this [Rust worker](https://github.com/supabase/walrus/tree/worker/worker) that improves the performance of filters and will revisit adding additional filters, and/or, etc.

**Comment by sampl** [comment]
> I honestly think the subscription version should be the exact duplicate of the query version

+1 for sure. Also, a lot of this issue seems very similar to "Query subscriptions" https://github.com/supabase/realtime-js/issues/221

**Comment by pedrovivaldi** [comment]
I could use this feature as well
